sympy/physics/units/dimensions.py

```diff
@@ -572,3 +572,127 @@ def is_consistent(self):
+            >>> L = length
+            >>> T = time
+            >>> I = current
+            >>> Lc = L # Characteristic length scale
+            >>> Uc = L/T # Characteristic velocity scale
+            >>> rho_c = M/L**3 # Characteristic density scale
+            >>> mu = M/L/T # Dynamic viscosity
+            >>> p_c = M/L/T**2 # Characteristic pressure scale
+            >>> F_bc = M*L/T**2 # Characteristic body forces scale
+            >>> dict_of_quantities = {'Lc': Lc, 'Uc': Uc, 'rho': rho_c, 'mu': mu, 'pc': p_c, 'Fbc': F_bc}
+            >>> dimsys_SI.get_dimensionless_numbers(dict_of_quantities)
+            [Dimension(1), Dimension(1), Dimension(1)]
+            >>> set_of_dimless_nums = dimsys_SI.get_dimensionless_numbers(dict_of_quantities)
+            >>> flag = True
+            >>> for dimension in set_of_dimless_nums:
+            ...     flag = flag and dimsys_SI.get_dimensional_dependencies(dimension) == {}
+            >>> flag
+            True
+            >>> dimsys_SI.print_dimensionless_numbers(dict_of_quantities)
+            ['[Lc]**(-1)*[Uc]**(-1)*[rho]**(-1)*[mu]**(1)*[pc]**(0)*[Fbc]**(0)', '[Lc]**(0)*[Uc]**(-2)*[rho]**(-1)*[mu]**(0)*[pc]**(1)*[Fbc]**(0)', '[Lc]**(-2)*[Uc]**(-2)*[rho]**(-1)*[mu]**(0)*[pc]**(0)*[Fbc]**(1)']
+            >>> # The first line of the output is the Reynolds number Re = mu/(Uc*Lc*rho)
+            >>> # The second line of the output is the Euler number Eu = pc/(rho*Uc**2)
+            >>> # The third line of the output is the Froude number Fr, such that Fr**2 = Fbc/(rho*Uc**2*Lc**2)
+
+        As an another example, consider the scenario in the case of a magnetodynamics problem governed by
+        Navier-Stokes equations coupled to Maxwell's equations for an incompressible
+        Newtonian fluid in an CGS (Gaussian) system ....
+
+            >>> from sympy.physics.units import length, mass, time, current
+            >>> from sympy.physics.units.systems.cgs import dimsys_cgs
+            >>> from sympy import S
+            >>> M = mass
+            >>> L = length
+            >>> T = time
+            >>> Lc = L # Characteristic length scale
+            >>> Uc = L/T # Characteristic velocity scale
+            >>> rho_c = M/L**3 # Characteristic density scale
+            >>> mu = M/L/T # Dynamic viscosity
+            >>> j0 = M**(S.One/2)/L**(S.One/2)/T**2 # Characteristic range value of current density
+            >>> sigma_0 = T**(-1) # Characteristic conductivity scale
+            >>> B0 = L**(-S.One/2)*M**(S.One/2)/T # Magnetic field characteristic scale
+            >>> dict_of_quantities = {'Lc': Lc, 'Uc' : Uc, 'rho': rho_c, 'mu': mu, 'j0': j0, 'sigma_0': sigma_0, 'B0': B0}
+            >>> dimsys_cgs.get_dimensionless_numbers(dict_of_quantities)
+            [Dimension(1),
+             Dimension(sqrt(mass)/(length**(3/2)*sqrt(mass/length**3))),
+             Dimension(1),
+             Dimension(sqrt(mass)/(length**(3/2)*sqrt(mass/length**3)))]
```

Comment:
The input is a dictionary while the output is a list. That does not seem very consistent.